### ISSUE TITLE
Edge functions deploy not working on m2 chip

### ISSUE THREAD

**Issue by holdehn** [supabase/cli#1266]
**Describe the bug**
When attempting to deploy a function via the Supabase CLI, I consistently receive an error message: "assertion failed [block != nullptr]: BasicBlock requested for unrecognized address (BuilderBase.h:550 block_for_offset)". This occurs regardless of whether I've installed Supabase using Homebrew or Node, and across different terminals (including the regular terminal and VS Code). I have ensured that my system architecture matches that of Node.js and followed all the troubleshooting steps suggested in [this Ethereum GitHub thread](https://github.com/ethereum/solidity/issues/13523#issuecomment-1544599434) to no avail.

**To Reproduce**

Run the following command in the terminal: supabase functions deploy [FUNCTION_NAME]
The error message appears during the bundling process.
Expected behavior
I expected the function to bundle and deploy without any errors.

**Screenshots**

<img width="718" alt="Screenshot 2023-06-29 at 12 46 42 PM" src="https://github.com/supabase/cli/assets/109185265/31b68eda-e303-406d-9eea-d505487e1be8">

**Desktop:**

OS: macOS 13.2.1, m2 chip 16gb ram
Version of CLI: v1.75.2
Version of Node.js: v18.16.1

**Additional context:**
This issue seems to be linked to an assertion error in the LLVM backend, as discussed in the Ethereum GitHub repository. I've made sure that my system is using the arm64 architecture, matching the Node.js installation, as suggested in the aforementioned thread. However, the error persists. All other commands work as expected, and the error is only triggered when attempting to bundle and deploy the specific function. I've also tried running the command with the --debug flag, but it does not provide additional insights.

**Comment by sweatybridge** [comment]
May I check what's the output of `deno --version` command? If it says x86, you may be able to fix it by deleting the deno binary and re-download. 

```bash
rm ~/.supabase/deno
supabase functions deploy
```

**Comment by holdehn** [comment]
%deno --version

deno 1.34.3 (release, aarch64-apple-darwin)
v8 11.5.150.2
typescript 5.0.4




it is working now! yay! I wish I had a better explanation for what I did.

You can see the error and the output of deno -v

<img width="682" alt="Screenshot 2023-06-30 at 9 59 56 AM" src="https://github.com/supabase/cli/assets/109185265/9a049d11-7950-4203-b66e-c701ee4fe3ea">

Then it started working right afterwards.

I think the issue was with clearing the deno cache and maybe specifying homebrew's path in vscode settings. 

also before I used "supabase functions deploy [FUNCTION_NAME]" to test, this time I used "supabase functions deploy" to deploy all functions at once, shouldn't make a difference but I thought it might be noteworthy.  The former works now too.

**Comment by bradydowling** [comment]
Thanks for posting this. Found the solution from one of the linked issues above, [namely this one](https://github.com/ethereum/solidity/issues/13523#issuecomment-1544599434). So for me, I'm pretty sure this was my process:

1. When I ran `arch` in VSCode, it showed `i386`
2. When I ran it in iTerm, it showed `arm64`, so I ran all my future commands from iTerm, just in case
3. `nvm uninstall <current_node_version>
4. `nvm install <current_node_version>
5. Run `node -e 'console.log(process.arch)'` and make sure it shows `arm64`
6. `rm ~/.supabase/deno`
7. Install supabase CLI via brew, at least version 1.75.5, [as mentioned here](https://github.com/supabase/supabase/issues/15312#issuecomment-1624241204) (I was using version 1.70.x locally via yarn before)

I also added this to my VSCode settings JSON, though I haven't tested out deploying from VSCode yet:

```
  "terminal.integrated.profiles.osx": {
    "bash": {
      "path": "arch",
      "args": ["-arm64", "bash", "-l"],
      "overrideName": true
    }
  },
  "terminal.integrated.defaultProfile.osx": "bash"
```
